lib/screenplay/screens_config/cache.ex

```diff
@@ -1,125 +1,12 @@
+    otp_app: :screenplay,
+    adapter: Nebulex.Adapters.Local
 
-  use Screenplay.Cache.Client, table: :screens_config
-
-  @type table_contents :: list(table_entry)
-
-  @type table_entry ::
-          {{:screen, screen_id :: String.t()}, ScreensConfig.Screen.t()}
-          | {:devops, ScreensConfig.Devops.t()}
-
-  def ok?, do: table_exists?()
-
-  def disabled?(screen_id) do
-    with_table default: false do
-      case :ets.match(@table, {{:screen, screen_id}, %{disabled: :"$1"}}) do
-        [[disabled]] -> disabled
-        [] -> false
-      end
-    end
-  end
-
-  def screen(screen_id) do
-    with_table default: nil do
-      case :ets.match(@table, {{:screen, screen_id}, :"$1"}) do
-        [[screen]] -> screen
-        [] -> nil
-      end
-    end
-  end
-
-  def app_params(screen_id) do
-    with_table default: nil do
-      case :ets.match(@table, {{:screen, screen_id}, %{app_params: :"$1"}}) do
-        [[app_params]] -> app_params
-        [] -> nil
-      end
-    end
-  end
-
-  def devops do
-    with_table default: nil do
-      case :ets.match(@table, {:devops, :"$1"}) do
-        [[devops]] -> devops
-        [] -> nil
-      end
-    end
-  end
-
-  @doc """
-  Returns a list of all screen IDs.
-  """
-  def screen_ids do
-    with_table default: [] do
-      @table
-      |> :ets.match({{:screen, :"$1"}, :_})
-      |> List.flatten()
-    end
-  end
-
-  @doc """
-  Returns a %{screen_id => screen_config} map of all screen configurations that satisfy the given filter.
-  The filter function will be passed a tuple of {screen_id, screen_config} and should return true if that screen should be included in the results.
-  """
-  def screens(filter_fn) do
-    with_table default: [] do
-      filter_reducer = fn
-        {{:screen, screen_id}, screen_config}, acc ->
-          if not screen_config.hidden_from_screenplay and filter_fn.({screen_id, screen_config}),
-            do: Map.put(acc, screen_id, screen_config),
-            else: acc
-
-        _, acc ->
-          acc
-      end
-
-      :ets.foldl(filter_reducer, %{}, @table)
-    end
-  end
-
-  @doc """
-  Gets the full map of screen configurations.
-
-  👉 WARNING: This function is expensive to run and returns a large amount of data.
-
-  Unless you really need to get the entire map, try to use one of the other client functions, or define a new one
-  that relies more on :ets.match / :ets.select to limit the size of data returned.
-  """
-  def screens do
-    with_table do
-      match_screen_entries = {{:screen, :"$1"}, :"$2"}
-      no_guards = []
-      output_entry_as_screen_id_screen_config_tuple = [{{:"$1", :"$2"}}]
-
-      match_spec = [
-        {match_screen_entries, no_guards, output_entry_as_screen_id_screen_config_tuple}
-      ]
-
-      @table
-      |> :ets.select(match_spec)
-      |> Map.new()
-    end
-  end
-
-  @doc """
-  Gets the entire config struct.
-
-  👉 WARNING: This function is expensive to run and returns a large amount of data.
```

Comment:
Given the new solution just always does this, unless fetching a specific screen ID, did this comment turn out to be incorrect?

Reply:
That was my conclusion. May have been an overly cautious warning. Each call to the function is nearly instant so there is no need for concern when retrieving the full cache.

Reply:
My intuition is also that this shouldn't have been much of a performance concern, so thanks for verifying.